test/WorkerExtensionTests/Blob/BlobStorageConverterTests.cs

```diff
@@ -22,375 +24,773 @@
+            var context = new TestConverterContext(typeof(BlobClient), grpcModelBindingData);
 
-            var conversionResult = await _mockBlobStorageConverter.Object.ConvertAsync(context);
+            var mockBlobClient = new Mock<BlobClient>();
+            mockBlobClient.Setup(m => m.Name).Returns("MyBlob");
 
-            Assert.Equal(ConversionStatus.Unhandled, conversionResult.Status);
+            var mockContainer = new Mock<BlobContainerClient>();
+            mockContainer.Setup(m => m.GetBlobClient(It.IsAny<string>())).Returns(mockBlobClient.Object);
+
+            _mockBlobServiceClient.Setup(m => m.GetBlobContainerClient(It.IsAny<string>())).Returns(mockContainer.Object);
+
+            // Act
+            var conversionResult = await _blobStorageConverter.ConvertAsync(context);
+            var clientResult = (BlobClient)conversionResult.Value;
+
+            // Assert
+            Assert.Equal(ConversionStatus.Succeeded, conversionResult.Status);
+            Assert.Equal("MyBlob", clientResult.Name);
         }
 
-        [Fact]
-        public async Task ConvertAsync_SourceAsModelBindingData_ReturnsSuccess()
+        [Fact (Skip = "Fails: ChangeType doesn't work for mock objects 'Object must implement IConvertible'")]
+        public async Task ConvertAsync_ValidModelBindingData_BlobClientCollection_List_ReturnsSuccess()
         {
-            object source = GetTestGrpcModelBindingData(GetFullTestBinaryData());
-            var context = new TestConverterContext(typeof(string), source);
-            _mockBlobStorageConverter
-                .Setup(c => c.ToTargetTypeAsync(typeof(string), Constants.Connection, Constants.ContainerName, Constants.BlobName))
-                .ReturnsAsync("test");
+            // Arrange
+            var grpcModelBindingData = GetTestGrpcCollectionModelBindingData(GetTestBinaryData());
+            var context = new TestConverterContext(typeof(IEnumerable<BlobClient>), grpcModelBindingData);
+
+            var mockBlobClient = new Mock<BlobClient>();
+            mockBlobClient.Setup(m => m.Name).Returns("MyBlob");
+
+            var mockContainer = new Mock<BlobContainerClient>();
+            mockContainer.Setup(m => m.GetBlobClient(It.IsAny<string>())).Returns(mockBlobClient.Object);
+
+            _mockBlobServiceClient.Setup(m => m.GetBlobContainerClient(It.IsAny<string>())).Returns(mockContainer.Object);
+
+            // Act
+            var conversionResult = await _blobStorageConverter.ConvertAsync(context);
+            var clientResult = (IEnumerable<BlobClient>)conversionResult.Value;
+
+            // Assert
+            Assert.Equal(ConversionStatus.Succeeded, conversionResult.Status);
+            Assert.IsAssignableFrom<IEnumerable<BlobClient>>(clientResult);
+            Assert.Equal("MyBlob", clientResult.First().Name);
+        }
+
+        [Fact (Skip = "Fails: ChangeType doesn't work for mock objects 'Object must implement IConvertible'")]
```

Comment:
You can add extra interfaces and associated implementations to mock objects.

``` CSharp
TMock mock = new Mock<TMock>();
Mock<IConvertable> convertable = mock.As<IConvertable>();
convertable.Setup(...);
```

Reply:
I'm not sure I understand how this works xD Do you mean that TMock is BlobClient i.e.

```csharp
    var mockBlobClient = new Mock<BlobClient>();
    Mock<IConvertible> convertible = mockBlobClient.As<IConvertible>();
```

Or that I should be creating something like TMock and using that to setup the blobClient?

Reply:
Sorry - `TMock` is just a placeholder. So yes, your example is correct. `BlobClient` would be `TMock` in this case.